fuel-vm/src/interpreter/executors/opcodes_impl.rs

```diff
@@ -1358,6 +1360,37 @@ where
     }
 }
 
+impl<M, S, Tx, Ecal, V> Execute<M, S, Tx, Ecal, V> for fuel_asm::op::JAL
+where
+    M: Memory,
+    S: InterpreterStorage,
+    Tx: ExecutableTransaction,
+    Ecal: EcalHandler,
+    V: Verifier,
+{
+    fn execute(
+        self,
+        interpreter: &mut Interpreter<M, S, Tx, Ecal, V>,
+    ) -> IoResult<ExecuteState, S::DataError> {
+        interpreter.gas_charge(interpreter.gas_costs().jmp())?;
+        let (reg_ret_addr, reg_target, offset) = self.unpack();
+
+        // Storing return address to zero register discards it instead
+        // While we use saturating_add here, the PC shoudln't ever have values above the
+        // memory size anyway
+        let ret_addr =
+            interpreter.registers[RegId::PC].saturating_add(Instruction::SIZE as u64);
```

Comment:
Are we certain the interpreter will always have the `RegId::PC` register available? How do we maintain that invariant?

Reply:
Yep, we're certain. There are hundreds of tests in this repo which all rely on that fact. The field is defined as`registers: [Word; VM_REGISTER_COUNT]`, and PC along with every other system register has an index below `VM_REGISTER_COUNT`.

Reply:
That's good, thanks for the pointer! :pray: